### PR TITLE
chore: remove touch as dev dependency

### PR DESCRIPTION
**Pull request by Uzlopak**
Remove touch as dev dependency.

### CODE REVIEW

File: test/integration/helpers.js [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 const escapeRegExp = require('escape-string-regexp');
 const os = require('os');
-const fs = require('fs-extra');
+const fs = require('fs');
+const fsExtra = require('fs-extra');
```
Comment (voxpelli) [inline test/integration/helpers.js:6]:
```suggestion
const {
  utimesSync,
} = require('fs');
const {
  ensureDirSync,
  readFileSync,
} = require('fs-extra');
```

Thinking out loud: How about destructuring here?

Also: Doesn't `fs-extra` re-export everything from `fs`? What's the reason for introducing `fs` in addition to `fs-extra`?

**Review by voxpelli — COMMENTED** [review]
Some comments

Generally I like a focused module like `touch` more than a huge one like `fs-extra`, but since `touch` hasn't seen an update in 6 years but `fs-extra` has gotten one in the last year, then 👍 

I do think we can leave `fs-extra` being imported as `fs` though?

File: test/integration/helpers.js (hunk 1 above)
Comment (Uzlopak) [reply test/integration/helpers.js]: i didnt know it is re exporing. 
I want to make as few as possible changes.

PTAL

**Review by JoshuaKGoldberg — APPROVED** [review]
🔪

File: test/integration/helpers.js [hunk 2]
```diff
@@ -487,9 +488,13 @@ async function runMochaWatchJSONAsync(args, opts, change) {
...
   fs.ensureDirSync(path.dirname(filepath));
-  touch.sync(filepath);
+  try {
+    fs.utimesSync(filepath, touchRef, touchRef);
+  } catch (e) {
+    const fd = fs.openSync(filepath, 'a');
+    fs.closeSync(fd);
+  }
 }
-
```
Comment (JoshuaKGoldberg) [inline test/integration/helpers.js:492]: [Style] Was removing this blank line intentional? (not at all a blocker for me, though I do prefer it there a bit 😄)

File: test/integration/helpers.js [hunk 3]
```diff
@@ -487,9 +488,13 @@ async function runMochaWatchJSONAsync(args, opts, change) {
...
 function touchFile(filepath) {
   fs.ensureDirSync(path.dirname(filepath));
-  touch.sync(filepath);
+  try {
+    fs.utimesSync(filepath, touchRef, touchRef);
+  } catch (e) {
+    const fd = fs.openSync(filepath, 'a');
+    fs.closeSync(fd);
+  }
 }
```
Comment (Uzlopak) [inline test/integration/helpers.js:497]:
```suggestion
}

```

**Pull request merged**